### ISSUE TITLE
`.to_s`/`.inspect` on tuples are not reversable

### ISSUE THREAD

**Issue by ryanstout** [crystal-lang/crystal#10707]
I'm programmatically generating some crystal code from a set of hardcoded config files in order to embed the data in the program. (Without the need to ship the config files)

Using a tuple as a key for a hash requires a space after the hash's opening `{` (to prevent the grammar from thinking its a macro) However, when calling .to_s on a hash with a tuple key, the output does not have the space. I always like where .to_s or inspect in other languages creates code you can paste back in if needed.

Would it make since to add a space around the `{` and `}` of tuples when calling `.to_s`?

A quick example of the output:

```
a = { {1,2} => true }
puts a
# {{1, 2} => true}
```

Thanks!

**Comment by Blacksmoke16** [comment]
> I always like where .to_s or inspect in other languages creates code you can paste back in if needed.

Related: https://github.com/crystal-lang/crystal/issues/9966

**Comment by straight-shoota** [comment]
Yeah, I would expect that you could turn a hash with simple data structures into code like that. Avoiding macro interpolation makes sense.

I don't think we want every tuple or hash wrapped in white space, though. That's probably gonna trigger some OCD.

It should be easy to implement for the specific cases where this applies: a hash or tuple as first element in a hash or tuple. For symmetry, we could also consider last elements, but that's a bit more complex. Without an opening delimiter, the closing delimiter `}}` should be harmless.

**Comment by simon1tan1** [comment]
I disagree. I think this is doing exactly what it is supposed to do.
